pkg/ddl/schema_test.go

```diff
@@ -412,19 +413,38 @@ func TestRenameTableAutoIDs(t *testing.T) {
 	tk2.MustExec(`set @@session.innodb_lock_wait_timeout = 0`)
 	tk2.MustExec(`BEGIN`)
 	tk2.MustExec(`insert into t values (null, 4)`)
+
+	v1 := dom.InfoSchema().SchemaMetaVersion()
+
 	go func() {
 		alterChan <- tk1.ExecToErr(`rename table t to ` + dbName + `2.t2`)
 	}()
 	waitFor(11, "t", "running")
 	waitFor(4, "t", "public")
+
+	// ddl finish does not mean the infoschema loaded.
+	// when infoschema v1->v2 switch, it take more time, so we must wait to ensure
+	// the new infoschema is used.
+	for {
```

Comment:
Maybe we can use `require.Eventually` and add a final timeout?

Reply:
Done